src/rendergraph/common/rendergraph/material.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <memory>
+
+#include "backend/basematerial.h"
+#include "rendergraph/assert.h"
+#include "rendergraph/materialshader.h"
+#include "rendergraph/materialtype.h"
+#include "rendergraph/texture.h"
+#include "rendergraph/uniformscache.h"
+#include "rendergraph/uniformset.h"
+
+namespace rendergraph {
+class Material;
+} // namespace rendergraph
+
+class rendergraph::Material : public rendergraph::BaseMaterial {
+  public:
+    Material(const UniformSet& uniformSet);
+    virtual ~Material();
+
+    // see QSGMaterial::compare.
+    // TODO decide if this should be virtual. QSGMaterial::compare is virtual
+    // to concrete Material can implement a compare function, but in rendergraph
+    // we can compare the uniforms cache and texture already here, which seems
+    // sufficient.
+    int compare(const Material* pOther) const {
+        DEBUG_ASSERT(type() == pOther->type());
```

Comment:
Do we want to early return here in case this is not the same type? I guess we might want to do that if `m_uniformsCache` and `pOther->m_uniformsCache` have different size which could lead to out of bound access, right?